tools/harness-thci/OpenThread.py

```diff
@@ -1511,14 +1511,18 @@ def reset(self):
 
         start_time = time.time()
         while time.time() < start_time + timeout:
-            time.sleep(0.3)
+            time.sleep(0.5)
             if not self.IsBorderRouter:
                 self._disconnect()
                 self._connect()
             try:
                 self.__executeCommand('state', timeout=0.1)
                 break
             except Exception:
+                self.__restartAgentService()
+                time.sleep(2)
+                self.__sendCommand('factoryreset', expectEcho=False)
+                time.sleep(0.5)
```

Comment:
we are already `time.sleep(0.5)` at the beginning of the while loop ?

Reply:
When I added this, I was thinking that it might make sense to give a device that actually hit the exception a little extra time just in case it needed it (since it didn't work with the initial timing).  I am not having an issue with it right now, but I thought that if someone hit the exception, this would provide a little more of a safety net.  I was approaching this from the standpoint of the reset needing to be as foolproof as possible so as to not cause failed tests to bleed over into others causing a string of false failures.  I am certainly okay with removing this line if you think this is unnecessary.

Reply:
We actually sleep for 3 seconds in OTCI after `factoryreset` for safety. 

Another technique is that we can issue `state` command multiple times after `factoryreset` until the device is responsive again.

Reply:
I am amenable to any style that is most beneficial.  However, the only solution I tested through full TH runs is the above.  Are you advocating for one of these other choices?

Reply:
Safer to use a solution that is verified by full TH runs. Thanks.